LSPS4/README.md

```diff
@@ -0,0 +1,1923 @@
+    `ln_base_fee_msat` and `ln_prop_fee`, respectively.
+  * `cltv_expiry_delta` is `cltv_expiry_delta`.
+* `c` contains a `min_final_cltv_expiry_delta` that is at least
+  +2 higher than its normal setting.
+  * The client MUST impose its normal setting on actual received
+    payment.
+    For example, suppose the client normally uses a
+    `min_final_cltv_expiry_delta` of 144.
+    For a JIT Channels invoice, it sets it to 146 in the invoice,
+    but only still expects 144 when the payment reaches it
+    even for JIT Channels invoices.
+  * If the client intends to wait for confirmation before sending
+    the preimage (when `client_trusts_lsp` is not specified or
+    is `false`), it MUST increase the margin to 2 plus the number
+    of confirmations it intends to wait.
+    For example, if it intends to wait for 1 confirmation, it
+    should add +3 to this setting.
+
+All other fields are to be filled in by the client.
+
+> **Rationale** Bitcoin blocks may be mined at any time.
+> When a payer sends out a payment, it must "lock in" a specific
+> blockheight for the timeout of the HTLC arriving to the client,
+> but if a block is mined between when the payer sends out the
+> payment and when the client receives it, then the timeout may
+> now be below the normal `min_final_cltv_expiry_delta` of the
+> client.
+> BOLT requires that the payee fail this with
+> `incorrect_or_unknown_payment_details` with a `height` indicating
+> the new current blockheight.
+> The client can determine this from the notification that the
+> LSP sends to the client, and this specification includes an API
+> the client can use to fail that, however, there is still a race
+> condition if the incoming payment is used to pay for a new channel
+> open: between the time that the client checks the CLTV expiry,
+> and the time that the LSP is able to open the channel and send the
+> HTLC, a new block can be mined.
+> This is a time-of-check vs. time-of-use race condition.
+>
+> A client also cannot check if it already has enough inbound
+> liquidity to not buy a channel, and thus leave off the +2
+> margin in that case: by the time the sender actually gets a
+> payment to the client, the inbound liquidity might have changed
+> (the LSP might have closed a channel to the client, for example).
+> This is again a time-of-check vs. time-of-use race condition.
+>
+> By *always* adding a small +2 margin here, we greatly reduce the
+> incidence of such an accident.
+
+#### Forward Compatibility With Post-BOLT11 Payment Schemes
+
+Ultimately, the client MUST provide the information below to
+the payer:
+
+* The client node ID (or any public key whose private key the
+  client knows).
+* An SCID issued from `lsps4.reserve_scid`.
+  * How much the LSP wants to charge for forwards through that
+    SCID.
+* The LSP node ID.
+* The LSP-provided `cltv_expiry_delta`.
+* The expiry for the payment.
+* Whether it can use MPP or not.
```

Comment:
Why remove this?  Payer can only use MPP if payee (client in this context) supports it.  In theory a payee can be simplified to the point that it does not support MPP at all, so it should signal whether it supports it or not.  LSPs are forced to always consider the possibility of MPP, as this lets us also consider the case of multiple unrelated tiny payments by using all of them to fund a single channel open.

Reply:
Ah, I thought this was a remnant of the LSPS2 spec. If the client doesn't support MPP, he can totally put that in the invoice, but I didn't think it was relevant for this spec, because it works regardless of MPP signaling or not.